apps/condo/domains/common/utils/accessControl.js

```diff
@@ -0,0 +1,54 @@
+const gql = require('graphql-tag')
+const { capitalize, isUndefined, isNull, isBoolean } = require('lodash')
+const { isObject } = require('validate.js')
+const { throwForbiddenError } = require('./apolloErrorFormatter')
+
+function accessControl (schemaType, schemaName, schema){
+    if (schemaType === 'GQLListSchema' && isObject(schema.access)) {
+        Object.getOwnPropertyNames(schema.access).forEach(crudOperation => {
+            const originalAccess = schema.access[crudOperation]
+            if (isBoolean(originalAccess)){
+                return
+            }
+            schema.access[crudOperation] = function ({ authentication: { item: user } } ) {
+                if (user && user.permissions) {
+                    const thisPermissionName = `can${capitalize(crudOperation)}${schemaName}s`
```

Comment:
${schemaName}s Is not a common case. The keystone have the right model name for such cases